examples/test-exec-doc.yml

```diff
@@ -0,0 +1,153 @@
+# Name of the workflow
+name: Run Tests on Push to Main
+
+# Set permissions required for the workflow
+permissions:
+  contents: write           # Allows writing repository contents
+  pull-requests: write      # Allows creating and modifying pull requests
+  id-token: write           # Allows generating tokens for authentication
+  issues: write             # Allows creating and modifying issues
+
+# Define the events that trigger the workflow
+on:
+  push:
+    paths:
+      - '**/*.md'           # Trigger on push to any Markdown file
+  pull_request:
```

Comment:
Does triggering on any pull request with changes to markdown files create another pull request? Does this mean the pipeline will run twice for every push/pull request that gets created with markdown changes?

Reply:
good one - it is triggering on both and we primarily want it to be triggered on changes to markdown files and that too only in the branch from where it was triggered. removed the pull_request bit